vault/cluster.go

```diff
@@ -339,45 +339,67 @@ func (c *Core) stopClusterListener() {
 	c.logger.Info("core/stopClusterListener: success")
 }
 
-// ClusterTLSConfig generates a TLS configuration based on the local cluster
-// key and cert.
+// ClusterTLSConfig generates a TLS configuration based on the local/replicated
+// cluster key and cert.
 func (c *Core) ClusterTLSConfig() (*tls.Config, error) {
 	cluster, err := c.Cluster()
 	if err != nil {
 		return nil, err
 	}
 	if cluster == nil {
-		return nil, fmt.Errorf("cluster information is nil")
+		return nil, fmt.Errorf("local cluster information is nil")
 	}
 
 	// Prevent data races with the TLS parameters
 	c.clusterParamsLock.Lock()
 	defer c.clusterParamsLock.Unlock()
 
-	if c.localClusterCert == nil || len(c.localClusterCert) == 0 {
-		return nil, fmt.Errorf("cluster certificate is nil")
-	}
+	forwarding := c.localClusterCert != nil && len(c.localClusterCert) > 0
```

Comment:
Basically in the enterprise case we may need to continue anyways so we just make the rest of the function do the right thing so the control flow can stay the same.